lib/package.rb

```diff
@@ -131,10 +131,22 @@ def self.system(*args, **opt_args)
     end
 
     # after removing the env hash, all remaining args must be command args
-    cmd_args = args.join(' ')
+    cmd_args = args
 
-    # Add -j arg to build commands.
-    cmd_args.sub!(/\b(?<=make)(?=\b)/, " -j#{CREW_NPROC}") unless cmd_args =~ /-j\s*\d+|cmake/
+    # split cmdline to multi arguments if the command is passed in one single string
+    cmd_args = args[0].scan(/(?:\w|["'].*?["'])+/) if args.size == 1
+
+    # add -j arg to build commands
+    cmd_args.map! do |arg|
+      if arg == 'make'
+        [ arg, "-j#{CREW_NPROC}" ]
+      else
+        arg
+      end
+    end.flatten
+
+    # involve a shell if the command is passed in one single string
+    cmd_args = [ 'bash', '-c', cmd_args.join(' ') ] if args.size == 1
```

Comment:
args to `bash -c` need to be passed inside quotes.

Reply:
Not needed when the command is passed in multi parameters?
```ruby
Kernel.system 'bash', '-c', 'ls -la'
# equals to
Kernel.system "bash -c 'ls -la'"
```

Reply:
That sounds good. Do you remember what packages were causing a problem with this before? We should test on those packages...

Reply:
Still a problem.

Try `crew build qemu`. I get this:
```
Performing post-install for glib...
/usr/local/lib/crew/lib/package.rb:152:in `system': wrong first argument
glib failed to build: `CREW_DISABLE_ENV_OPTIONS="1" ["bash", "-c", "glib compile schemas usr local share glib 2 0 schemas"]` exited with 0
```